### ISSUE TITLE
BUG: series.pop accepts more types than it's type signature says

### ISSUE THREAD

**Issue by anilbey** [pandas-dev/pandas-stubs#627]
**Describe the bug**
Similar to https://github.com/pandas-dev/pandas-stubs/issues/592, Series.pop also works on Hashable types but the signature does not mention them:

```python
def pop(self, item: _str) -> Series[S1]: ...
```

**To Reproduce**

Here is the code. Running it works fine.

```python
from enum import Enum
import pandas as pd

class MyEnum(Enum):
    TAYYAR = "tayyar"
    HAYDAR = "haydar"

df = pd.DataFrame(data = [[12.2, 10], [8.8, 15], [22.1, 14]], columns=[MyEnum.TAYYAR, MyEnum.HAYDAR])
series = df.loc[0]
print(series.pop(MyEnum.TAYYAR))

```

 Calling it with mypy gives the following error.
```
error: Argument 1 to "pop" of "Series" has incompatible type "MyEnum"; expected "str"  [arg-type]
```

**Please complete the following information:**
 - OS: linux
 - python version: 3.9
 - version of type checker: mypy                 1.1.1
 - version of installed `pandas-stubs`: pandas-stubs         1.5.3.230321

**Comment by anilbey** [comment]
Similar to https://github.com/pandas-dev/pandas-stubs/pull/596, shall I create a PR to add Hashable to its signature and provide the tests?

**Comment by anilbey** [comment]
@Dr-Irv if possible, could you make a new release on pypi with this change?

**Comment by Dr-Irv** [comment]
> @Dr-Irv if possible, could you make a new release on pypi with this change?

The plan is to put in the changes described in #624 and then do a 2.0.0.yymmdd release.  That will depend on getting the 2.0 changes into the stubs.  

I may need to alter that plan, since recent PR's have fixed things for 1.5.3.

**Comment by anilbey** [comment]
Brilliant, thanks @Dr-Irv
